Add motivation for attribute vs. element (#613)
openui/open-ui: research/src/pages/popup/popup.research.explainer.mdx

## Why a content attribute?

Again, refer to the [Other Alternatives Considered](/components/popup.proposal.alternatives) document for an exhaustive look at the other alternatives. That document runs through a fairly complete design of the other alternatives, to see what they would look like.

This section simply tries to summarize the primary reason that a content attribute was chosen to enable the pop-up behavior: a content attribute allows *behavior* to be applied to *any element*. That is useful:

```html
<dialog popup=auto>
  This is a "light dismiss" dialog
  <button>Ok</button>
  <button>Cancel</button>
</dialog>
```

Here, the developer is building **a dialog**. Since HTML [strongly encourages](https://html.spec.whatwg.org/multipage/dom.html#semantics-2) the use of the correct element that properly represents the **semantics of the content**, the proper element in this case is a `<dialog>`.

```html
<menu popup=auto>
  <li><button>Item 1</button></li>
  <li><button>Item 2</button></li>
  <li><button>Item 3</button></li>
</menu>
```

> Similarly here, the developer is building **a menu**, so they should use the `<menu>` element. This is natural and expected.

Reviewer: ```suggestion
Similarly here, the developer is building **a menu**, so they should use the `<menu>` element. This is adheres to the semantic definition of a `<menu>` element, and allows the content to be programmatically exposed as a listing of buttons.
```

Author: Done.

In both cases, not only does the use of the semantically-correct element make it easier to understand the content, it also enforces the appropriate content rules and behaviors that are specific to each element. Further, using the correct element allows the UA to correctly represent the content in the accessibility tree, making assistive technologies better able to assist users in navigating the content.

> In an alternative proposal where the pop-up behavior is enabled via a special element, e.g. `<popup>`, all of the above would need to be carefully managed by the developer via ARIA roles and attributes:

Reviewer: ```suggestion
In an alternative proposal where the pop-up behavior is enabled via a special element, e.g. `<popup>`, all of the above would need to be carefully managed by the developer via ARIA roles and attributes as there is no single role that can consistently be used to identify a `<popup>` element:
```

Author: Done.

```html
<popup role=dialog>

> <popup role=menu>

Reviewer: correction here, because although one would think that `<menu>` == `role=menu`, the menu element is really just a semantically different way of specifying a `<ul>` elment.

```suggestion
<popup role=list>
```

Author: Ugh. Really? Ugh. Ok, done. 😢

Reviewer: that is the correct response.
